### DISCUSSION TITLE
gatsby-config require is only for the .js file in src/commands/develop.ts

### DISCUSSION THREAD

**Discussion by xelmido** [gatsbyjs/gatsby#36128] [Help]
### Preliminary Checks

- [x] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Hi, 

Just started to use GatsbyJS for a couple of weeks.

Just bumped up to a small issue while using the develop server in docker. I was trying to set the INTERNAL_STATUS_PORT through the .env file. But it was not being interpreted 

It seems the gatsby-config.ts if not being required in the gatsby develop command. What are your toughts about importing without an extension? 

### Reproduction Link

//

### Steps to Reproduce

1. Use only a gatsby-config.ts file 
2. Require the dotenv config 
```
require("dotenv").config({
  path: `.env.${process.env.NODE_ENV}`,
})
```
3. Add a INTERNAL_STATUS_PORT=443 to the .env.development file 
4. Start develop server  
5. Check /___services

### Expected Result

I expect the port is being set to 443

### Actual Result

It was a random port 

### Environment

```shell
System:
    OS: Linux 5.10 Alpine Linux
    CPU: (8) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Shell: 1.35.0 - /bin/ash
  Binaries:
    Node: 16.16.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.11.0 - /usr/local/bin/npm
  npmPackages:
    gatsby: ^4.17.0 => 4.18.2
    gatsby-plugin-google-analytics: ^4.17.0 => 4.18.0
    gatsby-plugin-image: ^2.17.0 => 2.18.1
    gatsby-plugin-manifest: ^4.17.0 => 4.18.1
    gatsby-plugin-react-helmet: ^5.17.0 => 5.18.0
    gatsby-plugin-sass: ^5.17.0 => 5.18.1
    gatsby-plugin-sharp: ^4.17.0 => 4.18.1
    gatsby-plugin-sitemap: ^5.17.0 => 5.18.1
    gatsby-source-filesystem: ^4.17.0 => 4.18.1
    gatsby-source-wordpress: ^6.17.0 => 6.18.1
    gatsby-transformer-json: ^4.17.0 => 4.18.0
    gatsby-transformer-sharp: ^4.17.0 => 4.18.1
  npmGlobalPackages:
    gatsby-cli: 4.18.1
```

**Comment by schwigri** [comment]
There is a previous issue on this: https://github.com/gatsbyjs/gatsby/issues/28495

Do you have `dotenv` installed? I don't see it in your list of packages; if not, please try `npm install --save dotenv` and retry.

**Comment by xelmido** [comment]
I have dotenv installed and I checked https://github.com/gatsbyjs/gatsby/issues/28495. 

The thing is; when i do it with .js. It works. But when using the config with typescript. It doesn't. 

I saw this had been fixed in february. Commit: 

https://github.com/gatsbyjs/gatsby/commit/04d1d37d53e28deb13ec46dd97fb79b2c6cc863e#diff-f6c9f1f4612126450c5f4faef41482818bca2bcf799366337167a43014167f4a

**Reply by schwigri** [reply]
I see, thanks for clarifying! I can confirm that I can reproduce this issue; it woks with `gatsby-config.js` but fails with `gatsby-config.ts`, so it must be related to how Gatsby is processing TypeScript config.

**Reply by xelmido** [reply]
@LekoArts, can you move this back to issues instead of help?

**Reply by LekoArts** [reply]
The `__services` are removed, there's nothing to set anymore
